### ISSUE TITLE
How to approach TTA with albumentations?

### ISSUE THREAD

**Issue by sakvaua** [albu/albumentations#136]
Hi.
Is there a recommended approach to do test time augmentation? I would like to pick several random augmentations and apply them to test and validation sets for later stacking. I'm not sure how to proceed. How do I make sure that one set of augmentations is applied to the whole test set and then to validation set and not random augmentations to every image?
Thanks.

**Comment by albu** [comment]
you can use `p=1` transforms or `always_apply` transforms if it's something like horizontal flip or vertical flip (determined except probability)

**Comment by BloodAxe** [comment]
@sakvaua Hi, albumentations does not support this by design. However this is popular task, so I take a chance and pr my library with handy PyTorch extensions including TTA: https://github.com/BloodAxe/pytorch-toolbelt/blob/develop/pytorch_toolbelt/inference/tta.py
